snapcraft_legacy/cli/_options.py

```diff
@@ -354,27 +363,30 @@ def get_build_provider_flags(build_provider: str, **kwargs) -> Dict[str, str]:
         raise RuntimeError(f"Invalid build provider: {build_provider}")
 
     for option in _PROVIDER_OPTIONS:
-        key: str = option["param_decls"]  # type: ignore
+        keys: List[str] = option["param_decls"]  # type: ignore
         is_flag: bool = option.get("is_flag", False)  # type: ignore
         envvar: Optional[str] = option.get("envvar")  # type: ignore
         supported_providers: List[str] = option["supported_providers"]  # type: ignore
 
-        # Skip --provider option.
-        if key == "--provider":
-            continue
+        for key in keys:
+            # TODO: skip single character flags (e.g. keep `verbose` but discard `v`)
```

Comment:
This new code has a side effect of adding both `v` and `verbose` to the `build_provider_flags`. I think this is harmless, but the proper solution would be to discard any single character flags (keep verbose and discard v). 

I can add this on Monday to this PR or follow up in a subsequent PR.

Reply:
`-v` and `--verbose` are not actually being added to `build_provider_flags`.  Added a docstring to clarify this in https://github.com/snapcore/snapcraft/pull/4025.